File: src/main/java/be/kdg/kandoe/backend/model/organizations/Category.java
package be.kdg.kandoe.backend.model.organizations;

import be.kdg.kandoe.backend.model.cards.CardDetails;
import lombok.*;

import javax.persistence.*;
import java.util.List;
import java.util.Set;

@Data
@Entity
@NoArgsConstructor
@EqualsAndHashCode(exclude = {"organization", "tags", "cards", "topics"})
public class Category {
    @Id
    @GeneratedValue
    @Setter(AccessLevel.NONE)
    private int categoryId;

    @Column(nullable = false)
    private String name;

    private String description;

    @ManyToOne(optional = false)
    private Organization organization;

    @OneToMany
    private List<Tag> tags;
    
    //@OneToMany(targetEntity = CardDetails.class, fetch = FetchType.EAGER, cascade = { CascadeType.PERSIST, CascadeType.REFRESH, CascadeType.MERGE })
    @OneToMany(targetEntity = CardDetails.class, fetch = FetchType.EAGER, mappedBy = "category"/*, cascade = CascadeType.ALL*/) // mappedBy is needed to prevent unique key violation!
    private Set<CardDetails> cards;

    @OneToMany
    private List<Topic> topics;
}

File: src/main/webapp/app/entities/category.ts
import {Serializable} from "../util/serializable";

export class Category implements Serializable<Category>{
    public name : string;
    public description: string;

    constructor(name:string, description:string) {
        this.name = name;
        this.description = description;
    }


    public static createEmptyCategory():Category {
        return new Category("","");
    }

    deserialize(object:Category):Category {
        this.name = object.name;
        this.description = object.description;

        return this;
    }
}